pkg/policy/resolve.go

```diff
@@ -587,32 +596,32 @@ func (l4policy L4DirectionPolicy) forEachRedirectFilter(yield func(*L4Filter, Pe
 	return ok
 }
 
-// ConsumeMapChanges applies accumulated MapChanges to EndpointPolicy 'p' and returns a symmary of changes.
-// Caller is responsible for calling the returned 'closer' to release resources held for the new version!
-// 'closer' may not be called while selector cache is locked!
+// ConsumeMapChanges applies accumulated MapChanges to EndpointPolicy 'p' and returns a summary of
+// changes.  Caller is responsible for calling the returned 'closer' to release resources held for
+// the new revision!
 func (p *EndpointPolicy) ConsumeMapChanges() (closer func(), changes ChangeState) {
 	features := p.SelectorPolicy.L4Policy.Ingress.features | p.SelectorPolicy.L4Policy.Egress.features
-	version, changes := p.policyMapChanges.consumeMapChanges(p, features)
+	txn, changes := p.policyMapChanges.consumeMapChanges(p, features)
 
 	closer = func() {}
-	if version.IsValid() {
+	if txn.IsOpen() {
 		var msg string
-		// update the version handle in p.VersionHandle so that any follow-on processing
-		// acts on the basis of the new version
-		if p.VersionHandle.IsValid() {
-			p.VersionHandle.Close()
+		// update p.readTxn so that any follow-on processing acts on the basis of the new
+		// version
+		if p.readTxn.IsOpen() {
```

Comment:
I'm confused about the difference between `txn` and `p.readTxn` here. What do they mean?